verifier/verifier.go

```diff
@@ -518,6 +536,69 @@ func verifyAuthenticTimestamp(outcome *notation.VerificationOutcome) *notation.V
 	}
 }
 
+func verifyRevocation(outcome *notation.VerificationOutcome, client *http.Client, logger log.Logger) *notation.ValidationResult {
+	r, err := revocation.New(client)
+	if err != nil {
+		logger.Debug("error while calling revocation.New: %s", err.Error())
+		return &notation.ValidationResult{
+			Type:   trustpolicy.TypeRevocation,
+			Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+			Error:  errors.New("unable to check revocation status"),
+		}
+	}
+
+	signingTime := outcome.EnvelopeContent.SignerInfo.SignedAttributes.SigningTime
+	revocationResult := r.Validate(outcome.EnvelopeContent.SignerInfo.CertificateChain, signingTime)
+
+	result := &notation.ValidationResult{
+		Type:   trustpolicy.TypeRevocation,
+		Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+	}
+
+	currResult := revocation_base.ResultOK
```

Comment:
Why would we default to Unknown? Should we not start with OK and convert to it to Unknown when a certificate has that result? It seems weird to specify Unknown as the default, then check to see if every cert is OK or NonRevokable to eventually turn it back. There should not be any cases where this default is returned without being overridden by an Unknown or Revoked result that is present

Reply:
>There should not be any cases where this default is returned without being overridden by an Unknown or Revoked result that is present

This can happen during code bug and thats the reason its recommended to start with non-happy case to avoid such issues.

Reply:
But what I'm saying is how can this ever not be representative of the results? It goes through every result and checks the status, so if any are not OK it will reassign the currResult. I understand the hesitation, but I also think that starting with ResultUnknown makes it much more difficult to confirm that all of the results are OK if there is no error.

For example, the alternative to this would be going through the same logic but starting with a ResultUnknown. Then, we would need to count the number of OK results and compare to the total number of results in order to guarantee that they are all OK. I think this approach is much less understandable, but I can implement it if you think it is a better approach. I just don't see a scenario where this could be an issue. Can you give me a potential case where this Result would be incorrect because of the default?

Reply:
Updated to make logic more clear and concise and default to ResultUnknown, since false negative is preferred over false positive in the case of a bug/unexpected error.